packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol

```diff
@@ -237,4 +228,136 @@ contract L2CrossDomainMessenger_Test is Bridge_Initializer {
         assertEq(l2CrossDomainMessenger.successfulMessages(hash), true);
         assertEq(l2CrossDomainMessenger.failedMessages(hash), true);
     }
+
+    /// @dev Tests that gasPayingToken returns the correct values for ETH.
+    function test_gasPayingToken_ether_succeeds() external {
+        (address token, uint8 decimals) = l2CrossDomainMessenger.gasPayingToken();
+        assertEq(token, address(Constants.ETHER));
+        assertEq(decimals, 18);
+    }
+
+    /// @dev Tests that gasPayingToken returns the correct values for non-ETH tokens.
+    function test_gasPayingToken_nonEther_succeeds() external {
+        vm.mockCall(address(l1Block), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(2)));
+        (address token, uint8 decimals) = l2CrossDomainMessenger.gasPayingToken();
+        assertEq(token, address(1));
+        assertEq(decimals, 2);
+    }
+
+    /// @dev Tests that isCustomGasToken returns the correct value for ETH as the gas token.
+    function test_isCustomGasToken_ether_succeeds() external {
+        assertFalse(l2CrossDomainMessenger.isCustomGasToken());
```

Comment:
I think the spec needs updating to ensure that it clarifies the messengers have this abi